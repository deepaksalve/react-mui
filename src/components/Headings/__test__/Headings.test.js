import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import { Title, SubTitle } from "../";

describe("Headings", () => {
  describe("Title", () => {
    test("renders", () => {
      render(<Title>This is a title</Title>);

      const elm = screen.getByTestId("title");

      expect(elm).toBeInTheDocument();
    });

    test("should content correct text", () => {
      render(<Title>This is a title</Title>);

      const elm = screen.getByTestId("title");

      expect(elm).toHaveTextContent("This is a title");
    });

    test("matches snapshot", () => {
      const tree = renderer.create(<Title>This is a title</Title>).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe("SubTitle", () => {
    test("renders", () => {
      render(<SubTitle>This is a subtitle</SubTitle>);

      const elm = screen.getByTestId("subtitle");

      expect(elm).toBeInTheDocument();
    });

    test("should content correct text", () => {
      render(<SubTitle>This is a subtitle</SubTitle>);

      const elm = screen.getByTestId("subtitle");

      expect(elm).toHaveTextContent("This is a subtitle");
    });

    test("matches snapshot", () => {
      const tree = renderer
        .create(<SubTitle>This is a subtitle</SubTitle>)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
