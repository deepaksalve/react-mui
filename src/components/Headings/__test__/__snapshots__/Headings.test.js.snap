// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headings SubTitle matches snapshot 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1 css-1vpzi8a-MuiTypography-root"
  data-testid="subtitle"
  style={
    Object {
      "color": "rgba(0, 0, 0, 0.6)",
    }
  }
>
  This is a subtitle
</p>
`;

exports[`Headings Title matches snapshot 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1 css-1vpzi8a-MuiTypography-root"
  data-testid="title"
>
  This is a title
</p>
`;
